grinder/core.py

```diff
@@ -1055,6 +1168,50 @@ def __process_current_product_queries(self, product_info: dict) -> None:
                     current_host, query, product_info
                 )
 
+        # Masscan queries processor
+        len_of_masscan_settings = len(product_info.get("masscan_settings") or [])
+        for query_index, query_info in enumerate(
+            product_info.get("masscan_settings") or []
+        ):
+            if not query_info.get("hosts"):
+                print("Hosts field is empty, skip this search")
+                continue
+
+            query_info["hosts"] = str(ip_network(query_info.get("hosts"), False))
+
+            hosts = query_info.get("hosts")
+            top_ports = int(query_info.get("top-ports", 0))
```

Comment:
<img width="717" alt="bug-2" src="https://user-images.githubusercontent.com/24269213/85234317-b835f900-b436-11ea-8845-f2ceca4c959d.png">

```
        "masscan_settings": [
            {
                "hosts": "92.63.64.1/28",
                "ports": "",
                "rate": "1000",
                "top-ports": ""
            },
            {
                "hosts": "92.63.64.17/28",
                "ports": "",
                "rate": "1000",
                "top-ports": ""
            },
            {
                "hosts": "92.63.64.33/28",
                "ports": "",
                "rate": "1000",
                "top-ports": ""
            }
        ],
```

Reply:
It is impossible to run the masscan scan without the top-ports argument? Really? Is this bug or feature?

Reply:
Its a bug! So, if we type `"top-ports": ""` in query file this `top_ports = int(query_info.get("top-ports", 0))` return `""` and `int()` from `""` throw exception. Fixed